src/services/doppler-billing-api-client.ts

```diff
@@ -0,0 +1,106 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { AppSession } from './app-session';
+import { RefObject } from 'react';
+
+export interface DopplerBillingApiClient {
+  getInvoices(page: number, pageSize: number): Promise<ResultWithoutExpectedErrors<Invoices>>;
+}
+
+interface DopplerBillingApiConnectionData {
+  jwtToken: string;
+  idUser: string;
+}
+
+export interface Invoices {
+  items: Invoice[];
+  totalItems: number;
+}
+
+export interface Invoice {
+  accountId: string;
+  product: string;
+  date: string;
+  currency: string;
+  amount: number;
+  filename: string;
+}
+
+export class HttpDopplerBillingApiClient implements DopplerBillingApiClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getDopplerBillingApiConnectionData(): DopplerBillingApiConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (
+      !connectionData ||
+      connectionData.status !== 'authenticated' ||
+      !connectionData.jwtToken ||
+      !connectionData.userData
+    ) {
+      throw new Error('Doppler Billing API connection data is not available');
+    }
+    return {
+      jwtToken: connectionData.jwtToken,
+      idUser: connectionData ? connectionData.userData.user.idUser : '',
```

Comment:
In the first line, you assume that `connectionData` is defined, but not in the second one, which is fine?

```
      jwtToken: connectionData.jwtToken,
      //        ^^^^^^^^^^^^^^^ You are assuming that connectionData is truly
      idUser: connectionData ? connectionData.userData.user.idUser : '',
      //      ^^^^^^^^^^^^^^^^^ You are asking if connectionData is truly or falsy
```

Reply:
The validation of null in the second line is unnecessary because previously we checked it. I will remove it.